GFW/convert.py

```diff
@@ -0,0 +1,103 @@
+import os
+import time
+import xarray as xr
+import rioxarray as rxr
+from rasterio.enums import Resampling
+from rasterio import Affine
+from urllib.request import urlretrieve
+from osgeo import gdal, osr
+import numpy as np
+import cftime as cf
+import datetime
+
+# vars
+sdate = datetime.datetime(2020, 1, 1)
+edate = datetime.datetime(2050, 1, 1)
+github_path = 'https://github.com/rubisco-sfa/ILAMB-Data/blob/msteckler/GFW/convert.py'
+
+# data sources
+remote_data = 'https://www.arcgis.com/sharing/rest/content/items/f950ea7878e143258a495daddea90cc0/data'
+local_data = 'sequestration_rate_mean_aboveground_full_extent_Mg_C_ha_yr.tif'
+remote_u_data = 'https://www.arcgis.com/sharing/rest/content/items/d28470313b8e443aa90d5cbcd0f74163/data'
+local_u_data = 'sequestration_error_ratio_layer_in_full_extent.tif'
+
+# Open the input raster
+download_stamp = time.strftime('%Y-%m-%d', time.localtime(os.path.getmtime(local_data)))
+if not os.path.isfile(local_data):
+    urlretrieve(remote_data, local_data)
+data = rxr.open_rasterio(local_data, band_as_variable=True)
+
+# ensure data is not projected and is wgs84
+epsg_code = int(data.rio.crs.to_epsg())
+if epsg_code != 4326:
+    data = data.rio.reproject(crs='EPSG:4326')
+
+# resample to 0.5 degrees
+target_res = 0.5
+resampled_data = data.coarsen(x=(int(target_res / abs(data.rio.resolution()[0]))),
+                              y=(int(target_res / abs(data.rio.resolution()[1])))).mean()
+
+# rename variable and x/y dims to lat/lon
+ds = resampled_data.rename({'x': 'lon', 'y': 'lat', 'band_1':'c_uptake_rate'})
+
+# numpy array of time bounds
+tb_arr = np.asarray([
+    [cf.DatetimeNoLeap(sdate.year, sdate.month, sdate.day)],
+    [cf.DatetimeNoLeap(edate.year, edate.month, edate.day)]
+]).T
+
+# xarray data array of time bounds
+tb_da = xr.DataArray(tb_arr, dims=('time', 'nv'))
+
+# add time dimension and time bounds attribute
+ds = ds.expand_dims(time=tb_da.mean(dim='nv'))
+ds['time_bounds'] = tb_da
+
+# define attributes
+t_attrs = {
+    'long_name':'time',
+    'units':'days since 2020-01-01',
+    'calendar':'noleap',
+    'bounds':'time_bounds',
+}
+y_attrs = {
+    'long_name':'latitude',
+    'units':'degrees_north',
+}
+x_attrs = {
+    'long_name':'longitude',
+    'units':'degrees_east',
+}
+v_attrs = {
+    'long_name': 'mean aboveground biomass carbon accumulation rate',
+    'units': 'Mg ha-1 yr-1',
+    'ancillary_variables': 'error_ratio',
```

Comment:
For my own edification, what is an error ratio? Is this to be interpreted as uncertainty?

Reply:
Yes, it's uncertainty! Should I call the attribute "bounds" instead of "ancillary_variables?" 

Also here's a description from them: "Estimates the uncertainty in carbon accumulation potential from natural forest regrowth. Specifically, the uncertainty metric is calculated as the modeled sequestration rate (mean of 100 random forest models) divided by the standard error of the 100 random forest models. Uncertainty is presented as a ratio because areas with higher sequestration rates tend to have high standard errors for their sequestration rates; presenting error as a ratio standardizes the model error by the sequestration rate. Higher numbers represent greater uncertainty in the model."

Reply:
Oh and should I call it "uncertainty" instead of "error ratio?"

Reply:
Ok, yes I know understand. I am not certain what to do here. One question is: is it CF-compliant? I believe that what you have encoded does follow that standard. However, another question is: does ILAMB know what to do with it? I understand why the authors use ratios now, but I am thinking for our purposes it would be better as a standard error measure.

Reply:
And yes, the current method of implementing uncertainty in ILAMB is to provide "bounds" for the variable. This may be easier to chat about offline.